### PR TITLE
Generate changelog; add PR template

### PR DESCRIPTION
**Pull request by JordanMartinez**
**Description of the change**

We don't currently have a changelog in this repo. Let's make things easier on ourselves. This was copy-pasted from the `purescript` repo. I'm guessing we don't need a 'CONTRIBUTORS.md' file?

---

**Checklist:**

- [x] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
- [ ] Linked any existing issues or proposals that this pull request should close
- [ ] Updated or added relevant documentation
- [ ] Added a test for the contribution (if applicable)

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+**Description of the change**
+
+Clearly and concisely describe the purpose of the pull request. If this PR relates to an existing issue or change proposal, please link to it. Include any other background context that would help reviewers understand the motivation for this PR.
+
+---
+
+**Checklist:**
+
+- [ ] Added the change to the changelog's "Unreleased" section with a reference to this PR (e.g. "- Made a change (#0000)")
```
Comment (thomashoneyman) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: Maybe (#0000 by @<user>)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+- [ ] Added myself to CONTRIBUTORS.md (if this is my first contribution)
```
Comment (thomashoneyman) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: I think we can remove this line

**Review by thomashoneyman — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (JordanMartinez) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (JordanMartinez) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

**Pull request merged**